File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06IO2/BridgePatern/BridgePatern/ShapeCircle.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePatern
{
    public class ShapeCircle :Shape
    {
        private static Brush bg = Brushes.Red;
        private float x, y, r;
        public ShapeCircle(string text)
        {
            ParseParams(text, v => x = v, v => y = v, v => r = v);
        }
        public override void Draw(Graphics g)
        {
            g.FillEllipse(bg, x - r, y - r, 2 * r, 2 * r);
        }
        public override string ToXml()
        {
            return $"<circle x=\"{x}\" y=\"{y}\" r=\"{r}\" />";
        }
        public override string ToString()
        {
            return $"Circle(x:{x},y:{y},r:{r})";
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022s/IZ08IO3/BridgeTest/src/ShapeCircle.java

import java.awt.Color;
import java.awt.Graphics;

public class ShapeCircle extends Shape
{
   public ShapeCircle(String text)
   {
      super(text,3);
   }
   public int x() { return tb[0]; }
   public int y() { return tb[1]; }
   public int r() { return tb[2]; }

   @Override public void Draw(Graphics g)
   {
      g.setColor(Color.red);
      g.fillOval(x()-r(),y()-r(),2*r(),2*r());
   }

   @Override public String toXml()
   {
      return String.format("<Circle x=%d y=%d r=%d/>",x(),y(),r());
   }

   @Override public String toString()
   {
      return String.format("Circle(%d,%d,%d)",x(),y(),r());
   }
}
